functions/Get-DbaConfiguration.ps1

```diff
@@ -0,0 +1,108 @@
+This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+You should have received a copy of the GNU General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.	
+
+.LINK
+https://dbatools.io/Get-DbaConfiguration
+
+.EXAMPLE
+Get-DbaConfiguration -SqlServer localhost
+Returns server level configuration data on the localhost (ServerName, ConfigName, DisplayName, ConfiguredValue, CurrentlyRunningValue)
+
+.EXAMPLE
+Get-DbaConfiguration -SqlServer localhost -Detailed
+Returns detailed information on server level configuration data on the localhost (ServerName, ConfigName, DisplayName, Description, IsAdvanced, IsDynamic, MinValue, MaxValue, ConfiguredValue, CurrentlyRunningValue)
+
+.EXAMPLE
+'localhost','localhost\namedinstance' | Get-DbaConfiguration
+Returns system configuration information on multiple instances piped into the function
+
+#>
+	[CmdletBinding()]
+	Param (
+		[parameter(Position = 0, Mandatory = $true, ValueFromPipeline = $True)]
+		[Alias("ServerInstance", "SqlInstance", "SqlServers")]
+		[string[]]$SqlServer,
+		[System.Management.Automation.PSCredential]$SqlCredential,
+		[switch]$Detailed
+	)
+	BEGIN
+	{
+		$output = @()
+	}
+	PROCESS
+	{
+		FOREACH ($Server in $SqlServer)
+		{
+			TRY
+			{
+				$server = Connect-SqlServer -SqlServer $server -SqlCredential $sqlcredential
+			}
+			CATCH
+			{
+				Write-Verbose "Failed to connect to: $Server"
+				continue
+			}
+			
+			#Get a list of the configuration property parents, and exlude the Parent, Properties values
+			$objList = get-member -InputObject $server.Configuration -MemberType Property -Force | select Name | where { $_.Name -ne "Parent" -and $_.Name -ne "Properties" }
+			
+			#Iterate through the properties to get the configuration settings
+			foreach ($prop in $objList)
+			{
+				$PropInfo = $server.Configuration.$($prop.Name)
+				
+				#Ignores properties that are not valid on this version of SQL
+				if (!([string]::IsNullOrEmpty($PropInfo.RunValue)))
+				{
+					$output += [pscustomobject]@{
+						ServerName = $server.Name
+						ConfigName = $($prop.Name)
+						DisplayName = $PropInfo.DisplayName
+						Description = $PropInfo.Description
+						IsAdvanced = $PropInfo.IsAdvanced
+						IsDynamic = $PropInfo.IsDynamic
+						MinValue = $PropInfo.Minimum
+						MaxValue = $PropInfo.Maximum
+						ConfiguredValue = $PropInfo.ConfigValue
+						CurrentlyRunningValue = $PropInfo.RunValue
+					}
+				}
+			}
+			
+			if ($Detailed -eq $true)
+			{
+				$output | Sort-Object ServerName, DisplayName
```

Comment:
We must avoid sorting inside a function.
It comes with a performance penalty, 
maybe the user doesn't care about the order, or
worse: he'll pipe again to sort-object